doc/api/assert.md

```diff
@@ -2010,6 +2010,34 @@ argument, then `error` is assumed to be omitted and the string will be used for
 example in [`assert.throws()`][] carefully if using a string as the second
 argument gets considered.
 
+## `assert.snapshot(value[, name])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `value` {string} the value to snapshot
+* `name` {string} the name of snapshot.
+  in case order of snapshots is non-deterministic,
```

Comment:
why would ordering be non-determinstic??

Reply:
If you run concurrent tests for example?

Reply:
oof, that's a great reason to never do that :-/

Reply:
maybe we should just require the name then, rather than opening the footgun?

Reply:
maybe in a further iteration we can use `asyncLocalStorage` to automatically use the test name when running inside node test runner, but that should not happen in this iteration anyway

Reply:
That definitely sounds like a future improvement - I'm saying that in THIS iteration, the name should be strictly required, because then nobody can ever accidentally run into determinism issues. Later, if the name can be reliably inferred, then it can be made optional.

Reply:
I dont really have a strong opinion regarding making snapshot name required, but a lot of the feedback here was about keeping the API simple, so I'd like to know if anyone thinks name should be optional.
CC @BridgeAR @benjamingr @cjihrig

Reply:
"simple" means "everything is required". Optionality is complexity.

Reply:
Parallelism is per-file anyway isn't it? I don't understand why it's non-deterministic and i'm fine with landing this even with some non-deterministic behavior as experimental and to iterate.

Reply:
Snapshots, though, are written to the file system - if there's no chance of nondeterminism, this comment shouldn't be here; and if there is a change, we shouldn't be providing an API to users that' easily nondeterministic.

"experimental" doesn't really change anything in practice; if it ships, people will rely on it, and its experimental status eventually won't override that to allow breakage.

Reply:
@benjamingr concurrency can optionally also be per test - and `assert.snapshot` should work independently of how `node:test` test works

just as an example, this will break with any other test runner or program that runs things in parallel:
```js
describre({ concurrency: true }, () => {
	it('1', async () => {
		await setTimeout(random());
		assert.snapshot(thing);
	});

	it('2', async () => {
		await setTimeout(random());
		assert.snapshot(otherThing);
	});
});
```

Reply:
Ok, I'm not sure I agree how common/feasible of a footgun it is and it feels to me people would have to work pretty hard to run into it.

That said if the smallest unit of concurrency is the test (and not the file) I think appending the test name to the snapshot could be an easy fix then?

Reply:
sure - or, requiring users of this API to provide a name, and then we could make it be "the test name" in a followon.

The simplest API - which is appropriate for a new feature - is when everything is explicit and required, and nothing is implicit or optional.

Reply:
There were objections raised to that above I think?  I think using the test name is a good workaround to keep everyone content with the API.

Reply:
There's a lot of comments; can you link to it? I'd love to have those who objected explain why an implicit optional name is simpler than an explicit required one.